features/preferences/impl/src/test/kotlin/io/element/android/features/preferences/impl/root/PreferencesRootPresenterTest.kt

```diff
@@ -104,4 +89,60 @@ class PreferencesRootPresenterTest {
             assertThat(loadedState.snackbarMessage).isNull()
         }
     }
+
+    @Test
+    fun `present - developer settings is hidden by default in release builds`() = runTest {
+        val presenter = createPresenter(
+            showDeveloperSettingsProvider = ShowDeveloperSettingsProvider(aBuildMeta(BuildType.RELEASE))
+        )
+        moleculeFlow(RecompositionMode.Immediate) {
+            presenter.present()
+        }.test {
+            skipItems(1)
+            val loadedState = awaitItem()
+            assertThat(loadedState.showDeveloperSettings).isFalse()
+        }
+    }
+
+    @Test
+    fun `present - developer settings can be enabled in release builds`() = runTest {
+        val presenter = createPresenter(
+            showDeveloperSettingsProvider = ShowDeveloperSettingsProvider(aBuildMeta(BuildType.RELEASE))
+        )
+        moleculeFlow(RecompositionMode.Immediate) {
+            presenter.present()
+        }.test {
+            skipItems(1)
+            val loadedState = awaitItem()
+            assertThat(loadedState.showDeveloperSettings).isFalse()
+
+            repeat(times = ShowDeveloperSettingsProvider.DEVELOPER_SETTINGS_COUNTER) {
+                loadedState.eventSink(PreferencesRootEvents.EnableDeveloperSettingsClicked)
```

Comment:
Maybe add at the beginning of the repeat block:

`assertThat(awaitItem().showDeveloperSettings).isFalse()`